src/ejabberd_app.erl

```diff
@@ -90,18 +87,15 @@ do_start() ->
 %% before shutting down the processes of the application.
 prep_stop(_State) ->
     mongoose_deprecations:stop(),
-    TypedListeners = get_typed_listeners(),
-    suspend_listeners(TypedListeners),
-    StoppedCount = broadcast_c2s_shutdown_sup(),
-    StoppedCount2 = broadcast_c2s_shutdown_to_regular_c2s_connections(TypedListeners),
+    StoppedCount = mongoose_listener:suspend_listeners_and_shutdown_connections(),
```

Comment:
Hm... why not call it from `mongoose_listener:stop` if it's going to be called always? Anyway, the current version is ok as well.

Reply:
It was in the first version I did in the morning.
After I saw a test failing...
So, test is written to mock `mongoose_listener:stop` to test that logic to stop clients works independently from the logic that stops listeners.

So, I've decided to have them separate, to allow to test them separately too.